vktrace/vktrace_replay/vkreplay_vkreplay.cpp

```diff
@@ -2256,16 +2166,183 @@ VkResult vkReplay::manually_replay_vkAllocateMemory(packet_vkAllocateMemory *pPa
+                            FSEEK(tracefp, (long)portabilityTable[j] + sizeof(packetHeader2) + (long)gimrPacket.pMemoryRequirements,
+                                  SEEK_SET);
+                            FREAD(&memRequirements, sizeof(memRequirements), 1, tracefp);
+
+                            gimrSizeSum += memRequirements.size;
+                            if (memRequirements.alignment > alignment) alignment = memRequirements.alignment;
+                            break;
+                        }
+
+                        if (packetHeader1.packet_id == VKTRACE_TPI_VK_vkDestroyImage ||
+                            packetHeader1.packet_id == VKTRACE_TPI_VK_vkDestroyBuffer) {
+                            FREAD(&destroyImagePacket, sizeof(destroyImagePacket), 1, tracefp);
+                            if (destroyImagePacket.image == bimPacket.image) {
+                                // Found a destroy of this Buffer/Image, stop the back search.
+                                break;
+                            }
+                        }
+                        if (j == 0) {
+                            // Got to the beginning of the list w/o finding the vkGIMR/vkGBMR call
+                            break;
+                        }
+                    }
+                }
+                if (!found) {
+                    // Didn't find corresponding gimr call.
+                    // vkAllocateMemory and vkBind{Image|Buffer}Memory were called without first calling
+                    // vkGet{Image|Buffer}MemoryRequirements.
+                    // Call it now and save the data.
+                    VkMemoryRequirements mem_reqs;
+                    if (packetHeader1.packet_id == VKTRACE_TPI_VK_vkBindImageMemory) {
+                        m_vkFuncs.real_vkGetImageMemoryRequirements(remappedDevice, m_objMapper.remap_images(bimPacket.image),
+                                                                    &mem_reqs);
+                        replayGetImageMemoryRequirements[m_objMapper.remap_images(bimPacket.image)] = mem_reqs;
+                    } else {
+                        m_vkFuncs.real_vkGetBufferMemoryRequirements(
+                            remappedDevice, m_objMapper.remap_buffers((VkBuffer)bimPacket.image), &mem_reqs);
+                        replayGetBufferMemoryRequirements[m_objMapper.remap_buffers((VkBuffer)bimPacket.image)] = mem_reqs;
+                    }
+                }
+            }
+        }
+        FSEEK(tracefp, saveFilePos, SEEK_SET);
+
+        if (!foundBindMem) {
+            // Didn't find vkBind{Image|Buffer}Memory call for this vkAllocateMemory.
+            // This isn't an error (the memory is probably allocated but never used),
+            // so just use the index from the trace file and continue.
+            replayResult = m_vkFuncs.real_vkAllocateMemory(remappedDevice, pPacket->pAllocateInfo, NULL, &local_mem.replayGpuMem);
```

Comment:
This fseek can be eliminated - we just called fseek with the same args on line 2306. Plan to change this.